src/DoctrineAuditBundle/Controller/AuditController.php

```diff
@@ -24,6 +25,46 @@ public function listAuditsAction(): Response
         ]);
     }
 
+    /**
+     * @Route("/audit/revert/{id}/{object}/{entity}/{field}", name="dh_doctrine_audit_revert")
+     *
+     * @param $id
+     * @param $object
+     * @param $entity
+     * @param $field
+     *
+     * @return \Symfony\Component\HttpFoundation\RedirectResponse
+     */
+    public function revertEntityHistoryAction($id, $object, $entity, $field)
+    {
+        // get audit reader service
+        $reader = $this->container->get('dh_doctrine_audit.reader');
+        // $am = $this->container->get('dh_doctrine_audit.manager');
+        // get audit entity manager
+        $em = $this->container->get('doctrine.orm.default_entity_manager');
+
+        /** @var AuditEntry $entity_audit */
+        $entity_audit = $reader->getAudit(AuditHelper::paramToNamespace($entity), $id);
+        $audited_entity = AuditHelper::paramToNamespace($entity);
+        $current_entity = $em->getRepository($audited_entity)->find($object);
+
+        // Get all differences
+        $diffs = $entity_audit[0]->getDiffs();
+        // get field value to revert
+        $field_value = $diffs[$field]['old'];
+
+        $setMethod = "set{$field}";
+
+        $current_entity->{$setMethod}($field_value);
+
+        $em->persist($current_entity);
+        $em->flush();
```

Comment:
All this kind of stuff does not have to live in a controller, it should be in a method of the manager for example.